### PR TITLE
Update CI configuration, add bullseye and bookworm jobs

### PR DESCRIPTION
**Pull request by legoktm**
This is copy-pasted from securedrop-client, with steps that don't apply
here removed. Some of them, like isort and mypy, really should be added
to this repository.

### CODE REVIEW

**Comment by legoktm** [comment]
@gonzalo-bulnes do you understand why `check-testing-requirements-bullseye` is failing here? Is the dev-requirements really that badly out of sync?

**Comment by gonzalo-bulnes** [comment]
I can reproduce that diff locally, with Python 3.9. And the diff looks very similar (except for the initial comment with the Python version) with Python 3.7.

I observe that most packages seems to have changed version... and that `dev-requirements.in` doesn't pin version. Could that explain it?

**Comment by gonzalo-bulnes** [comment]
(rebased without changes)

**Comment by gonzalo-bulnes** [comment]
Based on the CI pipeline failures, I think that the next steps here are:

1. Set up `securedrop-builder/securedrop-export` in order to be able to update the wheels ([instructions](https://github.com/freedomofpress/securedrop-builder#updating-python-wheels)
   - This does require some initial setup, as the `sha256sum.txt{,.asc}` are currently missing.
   - This should allow us to pass the [`check-testing-requirements-{bullseye,bookworm}`](https://app.circleci.com/pipelines/github/freedomofpress/securedrop-export/306/workflows/e818ec7a-260c-441f-9f73-317d50c33837/jobs/971) jobs in CI.
2. Updating the dev dependencies should then make the security checks successful (#113)

I might be wrong here, @legoktm @rocodes some steps review / proof-reading would be useful before getting started! But this is what things look like to me.

**Comment by rocodes** [comment]
(I think we just need to add the `VERSION_CODE` script that's in the client repo, and then tweak the Makefile so that requirements files for bookworm and bullseye are distinct, and then we're good to go. Does that seem right to someone with a little more CI xp? Don't want to clobber your PR with nonsense @legoktm :upside_down_face: )

File: requirements/dev-bookworm-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,411 @@
+#
+# This file is autogenerated by pip-compile with python 3.10
```
Comment (gonzalo-bulnes) [inline requirements/dev-bookworm-requirements.txt:2]: This looks fine: 3.10 :white_check_mark:

File: requirements/dev-bullseye-requirements.txt [hunk 2]
```diff
@@ -0,0 +1,413 @@
+#
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (gonzalo-bulnes) [inline requirements/dev-bullseye-requirements.txt:2]: This looks fine too: 3.9 :white_check_mark:

File: requirements/requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.7
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (gonzalo-bulnes) [inline requirements/requirements.txt:2]: And this looks fine as well: 3.9 :white_check_mark:

**Comment by gonzalo-bulnes** [comment]
@rocodes and I will clen the branch before merging. Meanwhile, could you please adjust the repo settings @eloquence ? We want the Bullseye checks to be required, and the Bookworm checks to be optional. :detective:

**Review by gonzalo-bulnes — APPROVED** [review]
Three of us have looked into this PR, so I'm confident we're not missing anything obvious. The build is green now, I've squashed and rebased what @rocodes and @legoktm have set up.

**Comment by eloquence** [comment]
This is done! @gonzalo-bulnes Could you double-check that all the required jobs are marked required?

**Comment by gonzalo-bulnes** [comment]
Yes, they are! Thank you @eloquence !

**Pull request merged**